infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -177,5 +184,6 @@ public boolean apply(LogReplicationEntryMsg msg) throws ReplicationWriterExcepti
     public void reset(long snapshot, long ackTimestamp) {
```

Comment:
let us create a method resetWithConfigSync() which does the additional step of syncing with the registry table.  This method will get called from everywhere except the init in SinkManager.

This will avoid the redundant sync with the registry table.